### PR TITLE
Add Fronius solar PV inverter monitor app

### PR DESCRIPTION
**Pull request by nchieng**
# Description
Credit to @marcusb 
I ported your SolarEdge monitor app to work with Fronius inverters.
I hope you that is okay with you. If not, let me know.

Displays real-time solar panel production data from your Fronius inverter via its local API.

Make sure the Fronius API is enabled via the Technician menu.

fronius_monitor

# Copilot

### Summary
🌞📊🐍

This pull request adds a new applet called Fronius Monitor, which displays real-time solar panel production data from a Fronius inverter. It includes the main code file `fronius_monitor.star`, the manifest file `manifest.yaml`, and the README file `README.md`. The applet is written in Starlark and uses the Fronius API to fetch and render the data.

> _`Fronius Monitor`_
> _Shows solar power in spring_
> _Starlark applet code_

### Walkthrough
* Create a new applet for displaying solar panel data from a Fronius inverter ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182), [link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-c0c213bd0897ac488dbf254479634b7ec7716994099c967ce32829cc05ca9657R1-R8), [link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-5a005e5bec7f4f0b88a4830307d5551f39ebc1f13eed500a9ae65692af01b9e1R1-R8))
  * Write the main logic of the applet in `fronius_monitor.star` using Starlark ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182))
    * Import modules for HTTP requests, JSON parsing, caching, and rendering ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182))
    * Define constants for API endpoints, colors, fonts, and icons ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182))
    * Define functions for fetching, parsing, and formatting data from the inverter ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182))
    * Define variables for storing and caching the data and the error state ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182))
    * Return a render.Root object that contains the graphical elements of the applet, such as text, icons, and bars ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182))
    * Handle the configuration of the applet, such as the IP address and the refresh interval of the inverter ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182))
    * Handle the error handling of the API requests, such as timeouts, invalid responses, and missing data ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-27e90d201ca704eb2386cb35fee0eb827b797c35ecd4499fdc8a69c3ef221e6cR1-R182))
  * Add a manifest file for the applet in `manifest.yaml` using YAML ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-c0c213bd0897ac488dbf254479634b7ec7716994099c967ce32829cc05ca9657R1-R8))
    * Specify the ID, name, summary, description, author, file name, and package name of the applet ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-c0c213bd0897ac488dbf254479634b7ec7716994099c967ce32829cc05ca9657R1-R8))
  * Add a README file for the applet in `README.md` using Markdown ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-5a005e5bec7f4f0b88a4830307d5551f39ebc1f13eed500a9ae65692af01b9e1R1-R8))
    * Provide a brief introduction, a screenshot, and a disclaimer about the applet ([link](https://github.com/tidbyt/community/pull/1470/files?diff=unified&w=0#diff-5a005e5bec7f4f0b88a4830307d5551f39ebc1f13eed500a9ae65692af01b9e1R1-R8))

### CODE REVIEW

**Comment by nchieng** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by nchieng** [comment]
recheck

**Comment by marcusb** [comment]
> Credit to @marcusb
> I ported your SolarEdge monitor app to work with Fronius inverters.
> I hope you that is okay with you. If not, let me know.

Certainly ok with me (in line with the license of this repo).

File: apps/froniusmonitor/fronius_monitor.star [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+    data = o["Body"]["Data"]["Site"]
+    pv = data["P_PV"]
+    power_load = data["P_Load"]
+    grid = data["P_Grid"]
+
+    points = []
+    flows = []
+
+    img = SOLAR_PANEL_OFF if pv == None or pv == 0 else SOLAR_PANEL
```
Comment (marcusb) [inline apps/froniusmonitor/fronius_monitor.star:90]: is this a Copilot thing? would have written 
```
img = SOLAR_PANEL if pv else SOLAR_PANEL_OFF
```

Comment (nchieng) [reply apps/froniusmonitor/fronius_monitor.star:90]: I was a bit confused about this.

Fronius docs say:
> #this value is null if inverter is not running
image

But on my own inverter, at night with no generation, it was returning 0 values.

`if pv` would be `True` if `0`. So I felt it reasonable to handle both `None` and `0`

File: apps/froniusmonitor/fronius_monitor.star [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        return "0", "W"
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "inverter_ip",
+                name = "Inverter IP Address",
+                desc = "Local IP Address of the Fronius inverter",
```
Comment (matslina) [inline apps/froniusmonitor/fronius_monitor.star:176]: While Tidbyt devices run on your local WiFI, the apps themselves execute in our backend, which means they have no way to access local IPs. If that's the plan for this app, then I'm afraid it won't work as things currently stand.

Comment (nchieng) [reply apps/froniusmonitor/fronius_monitor.star:176]: Drat! Okay, thanks for confirming. I'll look into going down the Solar Web route.

**Pull request closed without merging**